### ISSUE TITLE
unknown option single-branch

### ISSUE THREAD

**Issue by andwhyisit** [andwn/marsdev#1]
git clone https://github.com/konamiman/sjasm --branch v0.39 --single-branch
error: unknown option `single-branch'

The --single-branch option only works in newer versions of git. You might need to note that somewhere.

**Comment by andwn** [comment]
Maybe it'll still work if I just remove it since the branch is already specified

**Comment by andwhyisit** [comment]
Marsdev hates me:

make[2]: Entering directory '/root/marsdev/sik-tools/mdtools/mdtiler/tool'
cc -Wall -O3 -s -std=c99   -c -o main.o main.c
cc -Wall -O3 -s -std=c99   -c -o tiles.o tiles.c
cc -Wall -O3 -s -std=c99   -c -o batch.o batch.c
cc -Wall -O3 -s -std=c99   -c -o bitmap.o bitmap.c
bitmap.c:30:17: error: png.h: No such file or directory
bitmap.c:39: warning: parameter names (without types) in function declaration
bitmap.c: In function â€˜load_bitmapâ€™:
bitmap.c:105: error: â€˜png_structpâ€™ undeclared (first use in this function)
bitmap.c:105: error: (Each undeclared identifier is reported only once
bitmap.c:105: error: for each function it appears in.)
bitmap.c:105: error: expected â€˜;â€™ before â€˜png_ptrâ€™
bitmap.c:107: error: â€˜png_ptrâ€™ undeclared (first use in this function)
bitmap.c:113: error: â€˜png_infopâ€™ undeclared (first use in this function)
bitmap.c:113: error: expected â€˜;â€™ before â€˜info_ptrâ€™
bitmap.c:114: error: â€˜info_ptrâ€™ undeclared (first use in this function)
bitmap.c:115: warning: implicit declaration of function â€˜png_destroy_read_structâ€™
bitmap.c:122: warning: implicit declaration of function â€˜png_jmpbufâ€™
bitmap.c:129: warning: implicit declaration of function â€˜png_set_read_fnâ€™
bitmap.c:133: warning: implicit declaration of function â€˜png_set_user_limitsâ€™
bitmap.c:136: warning: implicit declaration of function â€˜png_read_pngâ€™
bitmap.c:136: error: â€˜PNG_TRANSFORM_STRIP_16â€™ undeclared (first use in this function)
bitmap.c:137: error: â€˜PNG_TRANSFORM_PACKINGâ€™ undeclared (first use in this function)
bitmap.c:137: error: â€˜PNG_TRANSFORM_SHIFTâ€™ undeclared (first use in this function)
bitmap.c:138: error: â€˜PNG_TRANSFORM_STRIP_ALPHAâ€™ undeclared (first use in this function)
bitmap.c:141: error: â€˜png_byteppâ€™ undeclared (first use in this function)
bitmap.c:141: error: expected â€˜;â€™ before â€˜rowsâ€™
bitmap.c:144: warning: implicit declaration of function â€˜png_get_image_widthâ€™
bitmap.c:145: warning: implicit declaration of function â€˜png_get_image_heightâ€™
bitmap.c:146: warning: implicit declaration of function â€˜png_get_color_typeâ€™
bitmap.c:186: error: â€˜PNG_COLOR_TYPE_PALETTEâ€™ undeclared (first use in this function)
bitmap.c:188: error: â€˜rowsâ€™ undeclared (first use in this function)
bitmap.c:195: error: â€˜PNG_COLOR_TYPE_GRAYâ€™ undeclared (first use in this function)
bitmap.c:206: error: â€˜PNG_COLOR_TYPE_RGBâ€™ undeclared (first use in this function)
bitmap.c: At top level:
bitmap.c:234: error: expected â€˜)â€™ before â€˜png_ptrâ€™
make[2]: *** [bitmap.o] Error 1
make[2]: Leaving directory '/root/marsdev/sik-tools/mdtools/mdtiler/tool'
make[1]: *** [/root/mars/bin/mdtiler] Error 2
make[1]: Leaving directory '/root/marsdev/sik-tools'
make: *** [sik-tools] Error 2

**Comment by andwn** [comment]
I think you would be better off extracting the release tarball to your home directory, if libpng 1.6 development libraries are not available in your OS.

**Comment by andwn** [comment]
Also if you only care about Cave Story I don't think any of Sik's tools (or Flamewing's for that matter) are needed, just the SGDK ones.

So `make sgdk` (don't clean first or you will lose the toolchain you spent an hour building) and you should be done.

**Comment by andwhyisit** [comment]
I am now deeply regretting choosing Centos when I originally set up this server.

**Comment by andwn** [comment]
😬 

This should "just work" though, really
https://github.com/andwn/marsdev/releases/tag/2017.09.12

**Comment by andwhyisit** [comment]
I wish it did.

/root/mars/m68k-elf/bin/m68k-elf-gcc: /lib64/libc.so.6: version `GLIBC_2.14' not found (required by /root/mars/m68k-elf/bin/m68k-elf-gcc)

I can't upgrade it without breaking stuff, so I have to compile from source.

**Comment by andwhyisit** [comment]
Alright I managed to compile the important stuff.

**Comment by andwn** [comment]
3 months later I was able to try this on a CentOS VM from work. I got rid of "--single-branch" and made the flamewing + sik tools optional since they require "new" things and aren't needed to build SGDK/Cave Story/YM2017. 

The GLIBC_2.14 error also came up from trying the release due to the compiler that built it using a newer glibc than that on the system (built with GCC 7 but CentOS 6 and Debian 8 only have GCC 4.8 compatibility).

Apparently libc is backwards compatible:
https://stackoverflow.com/questions/11107263/how-compatible-are-different-versions-of-glibc

So what I am going to do in the future is be more conservative with the GCC version that builds releases. The toolchain will still use GCC 7.2.0 but I will build the GCC 7.2.0 with GCC 4.8.x, if that makes sense. Cross compiling is fun.
